### PR TITLE
Source Amazon Seller Partner: cleanup spec duplicates and order UI fields

### PR DESCRIPTION
**Pull request by lizdeika**
## What
1. Cleanup duplicate Connector Specifications
2. "order" Specification input fields in the UI

#### Community member or Airbyter
   
- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/SUMMARY.md`
    - [x] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
</p>
</details>

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the new connector version is published, connector version bumped in the seed directory as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</p>
</details>

<details><summary> <strong> Connector Generator </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

### CODE REVIEW

**Comment by lizdeika** [comment]
Unit tests:
```
Test session starts (platform: darwin, Python 3.9.8, pytest 6.2.5, pytest-sugar 0.9.4)
cachedir: .pytest_cache
rootdir: /Users/lizdeika/Projects/airbyte, configfile: pytest.ini
plugins: sugar-0.9.4, mock-3.6.1, timeout-1.4.2
collecting ...
 airbyte-integrations/connectors/source-amazon-seller-partner/unit_tests/test_repots_streams_rate_limits.py::test_reports_stream_should_retry ✓   33% ███▍
 airbyte-integrations/connectors/source-amazon-seller-partner/unit_tests/test_repots_streams_rate_limits.py::test_reports_stream_send_request ✓   67% ██████▋   {"type": "LOG", "log": {"level": "INFO", "message": "Backing off _send_request(...) for 5.0s (airbyte_cdk.sources.streams.http.exceptions.DefaultBackoffException)"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Caught retryable error '' after 1 tries. Waiting 5 seconds then retrying..."}}
{"type": "LOG", "log": {"level": "INFO", "message": "Backing off _send_request(...) for 10.0s (airbyte_cdk.sources.streams.http.exceptions.DefaultBackoffException)"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Caught retryable error '' after 2 tries. Waiting 10 seconds then retrying..."}}
{"type": "LOG", "log": {"level": "INFO", "message": "Backing off _send_request(...) for 20.0s (airbyte_cdk.sources.streams.http.exceptions.DefaultBackoffException)"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Caught retryable error '' after 3 tries. Waiting 20 seconds then retrying..."}}
{"type": "LOG", "log": {"level": "INFO", "message": "Backing off _send_request(...) for 40.0s (airbyte_cdk.sources.streams.http.exceptions.DefaultBackoffException)"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Caught retryable error '' after 4 tries. Waiting 40 seconds then retrying..."}}
{"type": "LOG", "log": {"level": "ERROR", "message": "Giving up _send_request(...) after 5 tries (airbyte_cdk.sources.streams.http.exceptions.DefaultBackoffException)"}}

 airbyte-integrations/connectors/source-amazon-seller-partner/unit_tests/test_repots_streams_rate_limits.py::test_reports_stream_send_request_backoff_exception ✓100% ██████████
======================================================================= warnings summary =======================================================================
source_amazon_seller_partner/streams.py:140
  /Users/lizdeika/Projects/airbyte/airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/streams.py:140: DeprecationWarning: Call to deprecated class NoAuth. (Set `authenticator=None` instead) -- Deprecated since version 0.1.20.
    authenticator: HttpAuthenticator = NoAuth(),

.venv/lib/python3.9/site-packages/deprecated/classic.py:173
airbyte-integrations/connectors/source-amazon-seller-partner/unit_tests/test_repots_streams_rate_limits.py::test_reports_stream_should_retry
airbyte-integrations/connectors/source-amazon-seller-partner/unit_tests/test_repots_streams_rate_limits.py::test_reports_stream_send_request
airbyte-integrations/connectors/source-amazon-seller-partner/unit_tests/test_repots_streams_rate_limits.py::test_reports_stream_send_request_backoff_exception
  /Users/lizdeika/Projects/airbyte/airbyte-integrations/connectors/source-amazon-seller-partner/.venv/lib/python3.9/site-packages/deprecated/classic.py:173: DeprecationWarning: Call to deprecated class HttpAuthenticator. (Use requests.auth.AuthBase instead) -- Deprecated since version 0.1.20.
    return old_new1(cls, *args, **kwargs)

airbyte-integrations/connectors/source-amazon-seller-partner/unit_tests/test_repots_streams_rate_limits.py::test_reports_stream_should_retry
airbyte-integrations/connectors/source-amazon-seller-partner/unit_tests/test_repots_streams_rate_limits.py::test_reports_stream_send_request
airbyte-integrations/connectors/source-amazon-seller-partner/unit_tests/test_repots_streams_rate_limits.py::test_reports_stream_send_request_backoff_exception
  /Users/lizdeika/Projects/airbyte/airbyte-integrations/connectors/source-amazon-seller-partner/unit_tests/test_repots_streams_rate_limits.py:29: DeprecationWarning: Call to deprecated class NoAuth. (Set `authenticator=None` instead) -- Deprecated since version 0.1.20.
    authenticator=NoAuth(),

-- Docs: https://docs.pytest.org/en/stable/warnings.html

Results (5.33s):
       3 passed
```

Integration tests:
```
Test session starts (platform: darwin, Python 3.9.8, pytest 6.2.5, pytest-sugar 0.9.4)
cachedir: .pytest_cache
rootdir: /Users/lizdeika/Projects/airbyte, configfile: pytest.ini
plugins: sugar-0.9.4, mock-3.6.1, timeout-1.4.2
collecting ...

Results (0.03s):
```

Acceptance tests:
```
Test session starts (platform: darwin, Python 3.9.8, pytest 6.2.5, pytest-sugar 0.9.4)
cachedir: .pytest_cache
rootdir: /Users/lizdeika/Projects/airbyte, configfile: pytest.ini
plugins: sugar-0.9.4, mock-3.6.1, timeout-1.4.2
collecting ... {"type": "LOG", "log": {"level": "INFO", "message": "Docker run: \nspec\ninput: /private/var/folders/pd/rd8643xs7xz4fnhp4f4wlnk40000gn/T/pytest-of-lizdeika/pytest-1/test_match_expected_inputs0_0/run_1/input\noutput: /private/var/folders/pd/rd8643xs7xz4fnhp4f4wlnk40000gn/T/pytest-of-lizdeika/pytest-1/test_match_expected_inputs0_0/run_1/output"}}

 airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py::TestSpec.test_match_expected[inputs0] ✓             11% █▎
 airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py::TestSpec.test_required[inputs0] ✓                   22% ██▎
 airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py::TestSpec.test_optional[inputs0] ✓                   33% ███▍
 airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py::TestSpec.test_has_secret[inputs0] ✓                 44% ████▌
 airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py::TestSpec.test_secret_never_in_the_output[inputs0] ✓ 56% █████▋    {"type": "LOG", "log": {"level": "INFO", "message": "Docker run: \nspec\ninput: /private/var/folders/pd/rd8643xs7xz4fnhp4f4wlnk40000gn/T/pytest-of-lizdeika/pytest-1/test_oauth_flow_parameters_inp0/run_1/input\noutput: /private/var/folders/pd/rd8643xs7xz4fnhp4f4wlnk40000gn/T/pytest-of-lizdeika/pytest-1/test_oauth_flow_parameters_inp0/run_1/output"}}

 airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py::TestSpec.test_oauth_flow_parameters[inputs0] ✓      67% ██████▋   {"type": "LOG", "log": {"level": "INFO", "message": "Docker run: \ncheck --config tap_config.json\ninput: /private/var/folders/pd/rd8643xs7xz4fnhp4f4wlnk40000gn/T/pytest-of-lizdeika/pytest-1/test_check_inputs0_0/run_1/input\noutput: /private/var/folders/pd/rd8643xs7xz4fnhp4f4wlnk40000gn/T/pytest-of-lizdeika/pytest-1/test_check_inputs0_0/run_1/output"}}

 airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py::TestConnection.test_check[inputs0] ✓                78% ███████▊  {"type": "LOG", "log": {"level": "INFO", "message": "Docker run: \ndiscover --config tap_config.json\ninput: /private/var/folders/pd/rd8643xs7xz4fnhp4f4wlnk40000gn/T/pytest-of-lizdeika/pytest-1/test_discover_inputs0_0/run_1/input\noutput: /private/var/folders/pd/rd8643xs7xz4fnhp4f4wlnk40000gn/T/pytest-of-lizdeika/pytest-1/test_discover_inputs0_0/run_1/output"}}

 airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py::TestDiscovery.test_discover[inputs0] ✓              89% ████████▉ {"type": "LOG", "log": {"level": "INFO", "message": "Docker run: \ndiscover --config tap_config.json\ninput: /private/var/folders/pd/rd8643xs7xz4fnhp4f4wlnk40000gn/T/pytest-of-lizdeika/pytest-1/test_defined_cursors_exist_in_0/run_1/input\noutput: /private/var/folders/pd/rd8643xs7xz4fnhp4f4wlnk40000gn/T/pytest-of-lizdeika/pytest-1/test_defined_cursors_exist_in_0/run_1/output"}}

 airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py::TestDiscovery.test_defined_cursors_exist_in_schema[inputs0] ✓100% ██████████
=================================================================== short test summary info ====================================================================
SKIPPED [1] ../../bases/source-acceptance-test/source_acceptance_test/plugin.py:56: Skipping TestBasicRead.test_read because not found in the config
SKIPPED [1] ../../bases/source-acceptance-test/source_acceptance_test/plugin.py:56: Skipping TestFullRefresh.test_sequential_reads because not found in the config
SKIPPED [1] ../../bases/source-acceptance-test/source_acceptance_test/plugin.py:56: Skipping TestIncremental.test_two_sequential_reads because not found in the config

Results (9.91s):
       9 passed
```

**Comment by harshithmullapudi** [comment]
Hey @lizdeika thanks for the contribution. We will review it soon

**Comment by harshithmullapudi** [comment]
Hey @htrueman can you take a look at this PR once ?

File: airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/source.py [hunk 1]
```diff
@@ -28,35 +26,12 @@
 )
 
 
-class ConnectorConfig(BaseModel):
```
Comment (htrueman) [inline airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/source.py:31]: Are there any issues with that? It seemed to work just fine.

File: airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/source.py [hunk 2]
```diff
@@ -88,7 +63,6 @@ def check_connection(self, logger: AirbyteLogger, config: Mapping[str, Any]) ->
         Show error message in case of request exception or unexpected response.
         """
         try:
-            config = ConnectorConfig.parse_obj(config)  # FIXME: this will be not need after we fix CDK
```
Comment (htrueman) [inline airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/source.py:91]: Actually, we wanted to make pydantic config as a default in future CDK releases. So what's the point of getting rid of it?

**Review by htrueman — CHANGES_REQUESTED** [review]
I don't get why do we need to remove the pydantic spec. Could you, please, clarify it for me?

**Comment by lizdeika** [comment]
> I don't get why do we need to remove the pydantic spec. Could you, please, clarify it for me?

Because it is duplicated in `spec.json` and in `source.py` while `spec.json` seems enough.
Also it has FIX ME lines are not needed when using only `spec.json`.
Documentation does not say one needs to duplicate spec.json and pydantic: https://docs.airbyte.io/connector-development/connector-specification-reference

File: airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/source.py (hunk 2 above)
Comment (lizdeika) [reply airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/source.py:91]: The point is to not do the same twice: in spec.json and here

**Comment by htrueman** [comment]
> > I don't get why do we need to remove the pydantic spec. Could you, please, clarify it for me?
> 
> Because it is duplicated in `spec.json` and in `source.py` while `spec.json` seems enough. Also it has FIX ME lines that are not needed when using only `spec.json`. Documentation does not say one needs to duplicate spec.json and pydantic: https://docs.airbyte.io/connector-development/connector-specification-reference

I've discussed this topic within the team. We decided that the best option there is to stay with pydantic spec. You could place `spec.json` to `integration_tests` dir or get rid of it completely.

**Comment by lizdeika** [comment]
> > > I don't get why do we need to remove the pydantic spec. Could you, please, clarify it for me?
> > 
> > 
> > Because it is duplicated in `spec.json` and in `source.py` while `spec.json` seems enough. Also it has FIX ME lines that are not needed when using only `spec.json`. Documentation does not say one needs to duplicate spec.json and pydantic: https://docs.airbyte.io/connector-development/connector-specification-reference
> 
> I've discussed this topic within the team. We decided that the best option there is to stay with pydantic spec. You could place `spec.json` to `integration_tests` dir or get rid of it completely.

I see. So to make sure I understand this correctly: 
Future versions of cdk will use spec from a source.py spec method using pydantic and will not use `spec.json` at all?

**Comment by htrueman** [comment]
> 

We're going to remove the test that verify that `spec.json` exists in source dir. Instead we would validate the `spec` command output.
So we may use either `spec.json` or pydantic spec for now.

It is going to be only pydantic spec acceptable in some future releases. But it's not in the our close scope of work.

**Pull request closed without merging**